### PR TITLE
Fixes #450: Adds new function for generating a shorter form prival string.

### LINKED ISSUE #450: priority to string function
A function exists to get the string representation of a prival, `stumpless_get_prival_string`, which returns a string version of the two enumerations used to create the prival. However, this is longer and more verbose than need be, and a more succinct representation would follow the priority string convention used by tools like [logger](https://man7.org/linux/man-pages/man1/logger.1.html), for example `local3.info`. You will write a second function that generates this shorter form string for a prival.

## General Approach
There are a few details left out of the following approach, for you to fill in as you encounter them. If you find you need help, please ask here or on the [project gitter](https://gitter.im/stumpless/community) and someone can help you get past the stumbling block.

First, read [this](https://github.com/goatshriek/stumpless/blob/latest/docs/development.md#adding-new-functions) section of the development documentation on adding new functions. Once you understand how this process works, start defining the new function.

Review the existing implementation of `stumpless_get_prival_string` in `src/prival.c`. Once you understand how the current implementation works, start adding the new functionality to the same file and the associated header, in a function named `stumpless_get_priority_string`.

Note that `stumpless_prival_from_string` accepts strings in the same format of `facility.severity`. You can reference this implementation for ideas on how to implement the new function.

Add tests for your new functionality in the existing test module `test/function/prival.cpp`, based on the tests already there.

### PR DESCRIPTION
**Pull request by Jackb18b**
Adds a new function for a condensed and logger-standard prival string.

### CODE REVIEW

File: src/prival.c [hunk 1]
```diff
@@ -58,6 +58,31 @@ stumpless_get_prival_string( int prival ) {
...
+   
+  size_t len_severity = strlen(severity);
+  size_t len_facility = strlen(facility);
+
+  // +1 for '.' formatting, +1 for termination
+  priority_string_size = ( len_severity + len_facility + 2);
+  priority_string = alloc_mem( priority_string_size );
+
+  memcpy( priority_string, severity , len_severity); 
+  memcpy( priority_string + len_severity, ".", 1);
```
Comment (goatshriek) [inline src/prival.c]: this can just be an assignment `priority_string[len_severity] = '.';` rather than a memcpy

File: src/prival.c [hunk 2]
```diff
@@ -58,6 +58,31 @@ stumpless_get_prival_string( int prival ) {
...
+  size_t len_facility = strlen(facility);
+
+  // +1 for '.' formatting, +1 for termination
+  priority_string_size = ( len_severity + len_facility + 2);
+  priority_string = alloc_mem( priority_string_size );
+
+  memcpy( priority_string, severity , len_severity); 
+  memcpy( priority_string + len_severity, ".", 1); 
+  memcpy( priority_string + len_severity + 1, facility, len_facility);
+  memcpy( priority_string + len_severity + 1 + len_facility, "\0", 1);
```
Comment (goatshriek) [inline src/prival.c]: this can be a simple assignment as well

File: test/function/prival.cpp [hunk 3]
```diff
@@ -121,4 +121,15 @@ namespace {
...
     free( ( void * ) result );
   }
+
+  TEST(GetPriorityString, ValidPrival) {
+    int prival;
+    const char *result;
+
+    prival = STUMPLESS_SEVERITY_ERR | STUMPLESS_FACILITY_USER;
+    result = stumpless_get_priority_string( prival );
+    EXPECT_STREQ( result, "STUMPLESS_SEVERITY_ERR.STUMPLESS_FACILITY_USER" );
```
Comment (goatshriek) [inline test/function/prival.cpp]: refer to the issue for the correct format this string should have. In this case, it should be `user.err`.

**Review by goatshriek — CHANGES_REQUESTED** [review]
This is off to a good start! You'll need to add your new function to `src/windows/stumpless.def` for the Windows tests to pass.

Review the issue and make sure that you understand the format. This is different from the `stumpless_get_prival_string` format: the `STUMPLESS` prefixes aren't needed and the string should be in lowercase.

**Comment by m3g4d1v3r** [comment]
Hello @Jackb18b  and @goatshriek.
Can I co-author this PR? If you don't mind, I can make some commits after the old ones in order to let the PR pass the tests.

**Comment by goatshriek** [comment]
With no response from the original author, I think you're safe to bring it across the finish line. Abandoned PRs are not uncommon, particularly around Hacktoberfest.

**Comment by m3g4d1v3r** [comment]
I've added the fixes on another PR (#476) due to privilege issues on the original fork.

**Comment by goatshriek** [comment]
This work was finished by #476. The original author here has been given co-author credit, as the work was based on this one.

**Pull request closed without merging**